optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/common/BavetTuple.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet.common;
+
+import java.util.Arrays;
+
+public interface BavetTuple {
+
+    BavetNode getNode();
+
+    Object[] getFacts();
+
+    default String getFactsString() {
+        return String.join(", ", Arrays.stream(getFacts())
```

Comment:
You can do the same, and avoid creating an array inbetween, by using `Collectors.joining()` on the stream.

Reply:
Done, thanks :)